Signal/src/views/OWSFlatButton.swift

```diff
@@ -0,0 +1,176 @@
+                             selector: Selector) -> OWSFlatButton {
+        let button = OWSFlatButton()
+        button.setTitle(title:title,
+                        font: font,
+                        titleColor: titleColor )
+        button.setBackgroundColors(backgroundColor)
+        button.useDefaultCornerRadius()
+        button.setSize(width:width, height:height)
+        button.addTarget(target:target, selector:selector)
+        return button
+    }
+
+    public class func button(title: String,
+                             titleColor: UIColor,
+                             backgroundColor: UIColor,
+                             width: CGFloat,
+                             height: CGFloat,
+                             target:Any,
+                             selector: Selector) -> OWSFlatButton {
+        return OWSFlatButton.button(title:title,
+                                    font:fontForHeight(height),
+                                    titleColor:titleColor,
+                                    backgroundColor:backgroundColor,
+                                    width:width,
+                                    height:height,
+                                    target:target,
+                                    selector:selector)
+    }
+
+    public class func button(title: String,
+                             font: UIFont,
+                             titleColor: UIColor,
+                             backgroundColor: UIColor,
+                             target:Any,
+                             selector: Selector) -> OWSFlatButton {
+        let button = OWSFlatButton()
+        button.setTitle(title:title,
+                        font: font,
+                        titleColor: titleColor )
+        button.setBackgroundColors(backgroundColor)
+        button.useDefaultCornerRadius()
+        button.addTarget(target:target, selector:selector)
+        return button
+    }
+
+    public class func fontForHeight(_ height: CGFloat) -> UIFont {
+        let fontPointSize = round(height * 0.45)
+        return UIFont.ows_mediumFont(withSize:fontPointSize)!
+    }
+
+    // MARK: Methods
+
+    public func setTitle(title: String, font: UIFont,
+                         titleColor: UIColor ) {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        button.setTitle(title, for: .normal)
+        button.setTitleColor(titleColor, for: .normal)
+        button.titleLabel!.font = font
+    }
+
+    public func setBackgroundColors(upColor: UIColor,
+                                    downColor: UIColor ) {
+        guard let button = self.button else {
+            owsFail("Missing button")
+            return
+        }
+        button.setBackgroundImage(UIImage(color:upColor), for: .normal)
+        button.setBackgroundImage(UIImage(color:downColor), for: .highlighted)
+    }
+
+    public func setBackgroundColors(_ backgroundColor: UIColor ) {
```

Comment:
At the call sites above, I didn't notice the pluralization of this method, and assumed we were just calling UIView's built in `setBackgrondColor:`. 

How about renaming to `setBackgroundColors(upColor:)` /`setBackgroundColorsWithUpColor:` to disambiguate.

Reply:
Nice idea.  I also added a `owsFail()` in the `setBackgroundColor:` property setter to catch slip ups.